GTMAppAuthSwift/Tests/GTMKeychainManagerTests.swift

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+import XCTest
+import AppAuthCore
+@testable import GTMAppAuthSwift
+
+class GTMKeychainManagerTests: XCTestCase {
+  private let testServiceProvider = "fooProvider"
+  private let testUserID = "fooUser"
+  private let testEmail = "foo@foo.com"
+  private let testKeychainItemName = "testName"
+  private let keychainHelper = KeychainHelperFake()
+  private var keychainManager: GTMKeychainManager {
+    GTMKeychainManager(keychainHelper: keychainHelper)
+  }
+  private var authorization: GTMAppAuthFetcherAuthorization {
+    GTMAppAuthFetcherAuthorization(
+      authState: OIDAuthState.testInstance(),
+      serviceProvider: testServiceProvider,
+      userID: testUserID,
+      userEmail: testEmail,
+      userEmailIsVerified: "y"
+    )
+  }
+
+  override func tearDown() {
+    super.tearDown()
+    keychainHelper.passwordStore.removeAll()
+    keychainHelper.useDataProtectionKeychain = false
+  }
+
+  func testSaveAuthorization() throws {
+    try keychainManager.save(authorization: authorization, with: testKeychainItemName)
```

Comment:
Can we verify there is data in passwordStore?

Reply:
I don't think this is necessary given that the fake (and the actual implementation) end up calling through to this:

```
func removePassword(service: String) throws {
    guard !service.isEmpty else { throw Error.noService }
    let keychainQuery = keychainQuery(service: service)
    let status = SecItemDelete(keychainQuery as CFDictionary)

    guard status == noErr else { throw Error.failedToDeletePassword }
}
```

If an error is produced, then the test will fail. I'd rather not test an implementation detail in the fake like `passwordStore`.